system/jhtree/ctfile.cpp

```diff
@@ -298,8 +295,78 @@ void CWriteNodeBase::write(IFileIOStream *out, CRC32 *crc)
 
 CWriteNode::CWriteNode(offset_t _fpos, CKeyHdr *_keyHdr, bool isLeafNode) : CWriteNodeBase(_fpos, _keyHdr)
 {
-    keyLen = keyHdr->getMaxKeyLength();
     hdr.nodeType = isLeafNode ? NodeLeaf : NodeBranch;
+}
+
+//=========================================================================================================
+
+CPOCWriteNode::CPOCWriteNode(offset_t _fpos, CKeyHdr *_keyHdr, bool isLeafNode) : CWriteNode(_fpos, _keyHdr, isLeafNode)
+{
+    keyedLen = keyHdr->getNodeKeyLength();
+    hdr.subType = SplitPayloadLeaf;
+    // Initialize keyBytes to reflect our additional header info, plus the final entry in the offsets table (added later)
+    keyPtr += sizeof(SplitNodeHdr);
+    hdr.keyBytes = sizeof(SplitNodeHdr) + sizeof(unsigned short);
+}
+
+CPOCWriteNode::~CPOCWriteNode()
+{
+}
+
+bool CPOCWriteNode::add(offset_t pos, const void *data, size32_t size, unsigned __int64 sequence)
```

Comment:
fileposition not being saved.  It could be appended to the data if the index stored file positions.  And then no special case when extracting.

Reply:
I need to talk to you about fileposition...